velox/common/base/Portability.h

```diff
@@ -52,42 +53,80 @@ namespace facebook::velox {
 #endif
 #endif
 
-/// Define tsan_atomic<T> to be std::atomic<T?> for tsan builds and
-/// T otherwise. This allows declaring variables like statistics
-/// counters that do not have to be exact nor have synchronized
-/// semantics. This deals with san errors while not incurring the
-/// bus lock overhead at run time in regular builds.
-#ifdef TSAN_BUILD
+/// tsan_atomic is relaxed atomic without use rmw operations.
+/// This allows declaring variables like statistics counters that do not have to
+/// be exact nor have synchronized semantics.
+/// TODO: Use different name.
 template <typename T>
-using tsan_atomic = std::atomic<T>;
+class tsan_atomic : protected std::atomic<T> {
+ public:
+  using std::atomic<T>::atomic;
 
-template <typename T>
-inline T tsanAtomicValue(const std::atomic<T>& x) {
-  return x;
-}
+  T operator=(T v) noexcept {
+    this->store(v, std::memory_order_relaxed);
+    return v;
+  }
 
-/// Lock guard in tsan build and no-op otherwise.
-template <typename T>
-using tsan_lock_guard = std::lock_guard<T>;
+  operator T() const noexcept {
+    return this->load(std::memory_order_relaxed);
+  }
 
-#else
+  T operator+=(T v) noexcept {
+    const auto oldValue = this->load(std::memory_order_relaxed);
+    const auto newValue = oldValue + v;
```

Comment:
maybe a n00b question, but isn't this the same as

`this->store(*this + v, std::memory_order_relaxed)`

And maybe add a comment explaining how this is unsafe (by design)?